test/migration_repos_workflow_spec.rb

```diff
@@ -247,6 +247,23 @@
       allow(Registration::UrlHelpers).to receive(:registration_url)
     end
 
+    context "no migration product found" do
+      before do
+        expect(Y2Packager::ProductUpgrade).to receive(:new_base_product).and_return(nil)
```

Comment:
Putting an `expect` call looks wrong to me. You can change it to `allow` and put an `expect` in the test where want to check for that.

Reply:
This `expect` is intended for the whole `context`, it means that "no migration product found" condition. And to make sure that context is really used I used `expect`.

Reply:
Ok, let's keep it. But to me, checking things in the "before" block is wrong.